app/controllers/concerns/authentication_and_sso_concerns.rb

```diff
@@ -130,6 +148,17 @@ def sso_cookie_content
     }
   end
 
+  def sso_saml_cookie_content
+    return nil if @current_user.blank?
```

Comment:
Whether you perform the refactor I suggested above or not, this could move up into the other method. I'll post a comment where.

Reply:
done